matrix_client/api.py

```diff
@@ -434,13 +434,20 @@ def _send(self, method, path, content=None, query_params={}, headers={},
         if headers["Content-Type"] == "application/json":
             content = json.dumps(content)
 
-        response = requests.request(
-            method, endpoint,
-            params=query_params,
-            data=content,
-            headers=headers,
-            verify=self.validate_cert
-        )
+        response = None
+        while True:
```

Comment:
Hmm, you might be right on this one. It probably would make more sense to make this optional.